tests/console/validate_zfcp.pm

```diff
@@ -0,0 +1,199 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate zfcp in the installed system:
+#          - Verification of FCP devices online
+#          - Verificaiton of FCP devices
+#          - Verification of LUNs visible as SCSI devices
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use utils qw(arrays_subset);
+
+# Get devices online.
+# For each physical adapter in the mainframe there is a corresponding FCP channel providing physical
+# connection to the SAN Fabric (multiple FCP channels are configured to increase the I/O bandwidth
+# and improve data availability).
+# HBAs are shown in the system as FCP devices
+sub get_fcp_devices_online {
+    my $test_data = get_test_suite_data();
+
+    # FCP devices (online and offline)
+    my $fcp_devices = $test_data->{zfcp}->{fcp_devices};
+    # filtering online devices
+    return [grep { $_->{attributes}->{online} } @{$fcp_devices}];
+}
+
+# Compare fcp devices
+sub verify_fcp_devices {
+    my (%args) = @_;
+
+    my $devs_expected = [map { $_->{fcp_channel} } @{$args{devs_expected}}];
+    my $devs_current  = $args{devs_current};
+    my $only_warning  = $args{only_warning};
+
+    my @dif = arrays_subset($devs_expected, $devs_current);
+    if (scalar @dif > 0) {
```

Comment:
you could simplify this by converting the `@dif` to scalar directly `$dif` and remove the `scalar` here

Reply:
I don't see big difference changing that line, but could you write your suggestion?, not sure I got it.

Reply:
`my $dif = arrays_subset($devs_expected, $devs_current);
    if ($dif > 0) {`

Reply:
right, I could save one line, not exactly as you described but I got the idea though ;)